### ISSUE TITLE
Allow Vite dev server mode to be set to a value other than development

### ISSUE THREAD

**Issue by Csszabi98** [brillout/vite-plugin-ssr#516]
### Description

I was trying to set vite dev server mode like:

```
  const viteDevServer = (
    await vite.createServer({
      mode: 'production',
      root,
      server: { middlewareMode: true },
    })
  )
```

### Error Stack

```shell
Error: [vite-plugin-ssr@0.4.48][Bug] You stumbled upon a bug in vite-plugin-ssr's source code. Reach out at https://github.com/brillout/vite-plugin-ssr/issues/new or https://discord.com/invite/qTq92FQzKb and include this error stack (the error stack is usually enough to fix the problem). A maintainer will fix the bug (usually under 24 hours). Don't hesitate to reach out as it makes vite-plugin-ssr more robust. 
    at getRuntimeConfig (/PROJECT_DIR/node_modules/vite-plugin-ssr/dist/cjs/node/globalContext/runtimeConfig.js:17:24)
    at getGlobalContext (/PROJECT_DIR/node_modules/vite-plugin-ssr/dist/cjs/node/globalContext.js:49:64)
    at initializePageContext (/PROJECT_DIR/node_modules/vite-plugin-ssr/dist/cjs/node/renderPage.js:125:70)
    at renderPage_ (/PROJECT_DIR/node_modules/vite-plugin-ssr/dist/cjs/node/renderPage.js:30:47)
    at renderPage (/PROJECT_DIR/node_modules/vite-plugin-ssr/dist/cjs/node/renderPage.js:155:22)
    at file:///PROJECT_DIR/server.js:125:31
```

**Comment by brillout** [comment]
Indeed I could reproduce.

Why do you want to use Vite's dev server in production mode? I can't think of a reason for doing that.

**Comment by Csszabi98** [comment]
The “prodcution” mode is just an examle, in reality this is the deployment environment. I wanted to use the mode property to load environment variables for different targets (dev, uat, prod, etc.) which in turn modify the behavior of the applovation (like the called apis). E.g.: —mode uat -> load .env.uat with vite watching the env files.

**Comment by Csszabi98** [comment]
Thank you for the clarification commit. If the —mode flag is expected to be development in all cases, I’ll load the env files manually. The only thing I loose is just watch mode, for them, but they are not frequently modified so I can live with that (or just copy different configs under .env.development.local which will have watch mode).

**Comment by brillout** [comment]
You're not supposed to use Vite's dev server in production.

**Comment by Csszabi98** [comment]
Yes, and that was not the intention. It was to test features and bugs with different content that is bound to individual environments while developing locally, hence the loading of env files. Thanks!

**Comment by brillout** [comment]
I see.

It's possible to make VPS work that way, but I consider this low priority as most test production in its entirety, not only env variables. Feel free to disagree with my prioritization. (Sponsoring would make me reconsider as well 😉).

Changing the label from `Bug` to `Enhancement`.

**Comment by vedranmviu** [comment]
This also fails when running tests.
I'm using Vitest. And vitest, in my case, uses the same `vite.config.ts` file as the dev and build scripts.
The problem is that the vitest sets the environment to `test`, and it breaks the build.
I'd hate to separate`vite.config.ts` into two files because of this.

**Comment by brillout** [comment]
Yea I saw that as well. I've a fix in mind. ETA today.

**Comment by brillout** [comment]
Released in `0.4.52`.

In case your company is up for it: https://github.com/sponsors/brillout.
